lib/gst/player/gstplayer.c

```diff
@@ -112,6 +122,9 @@ struct _GstPlayerPrivate
   GstPlayerState app_state;
   gint buffering;
 
+  GstPlayerMediaInfo  *media_info;  /* current media info */
+  GstPlayerMediaInfo  *media_info_copy; /* last emited media info */
```

Comment:
The app will get the media info in some callback like this

    static void media_info_updated_cb (GstPlayer * player, GstPlayerMediaInfo *info, gpointer user_data);

The app can use the info while inside the scope of that function, but not afterwards anymore. If it wants to store the info elsewhere, it must get another reference with g_object_ref() and later unref() it again.